controllers/auth.go

```diff
@@ -36,6 +36,15 @@ func AuthenticatedUser(c echo.Context) models.User {
 // AuthenticateHandler : manages user authentication
 func AuthenticateHandler(c echo.Context) error {
 	var u models.User
+	var vu models.User
+
+	// validate input
+	vu.Username = c.FormValue("username")
+	vu.Password = c.FormValue("password")
+
+	if err := vu.Validate(); err != nil {
+		return echo.NewHTTPError(400, err.Error())
```

Comment:
A 403 is typically sent when accessing a protected resource without being authorised to do so. 

Seems a 422 (Unprocessable Entity) might better suit this purpose. https://developer.github.com/v3/#client-errors